saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md

```diff
@@ -0,0 +1,111 @@
+
+
+## SpRL using Saul
+In this example, we implement the best performing algorithm for [task 3 of SemEval 2012](https://www.cs.york.ac.uk/semeval-2012/task3.html) proposed in [2] using Saul.
+
+### Data representation and preparation
+The SpRL data are in the form of annotated sentences, we need basic data structures to load these data and feed them to the Saul application. [`SpRLSentence`](../../../../../../../../java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/SpRLSentence.java) and [`SpRelation`](../../../../../../../../java/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/Triplet/SpRelation.java) classes used for this purpose. In order to benefit from all of the built-in features provided by `TextAnnotation`, we used `TextAnnotation.Sentence` to represent the sentences in the `SpRLSentence` class.
+The data sets are collections of xml files, so we used code generators to generate corresponding classes and a data reader for them.
+
+So, the next step is to load these files and convert them to a collection of `SpRLSentence`. In this step we split sentences in each document and convert them to `SpRLSentence` objects.
+
+```
+    val reader = new SpRLDataReader(path, classOf[SpRL2013Document])
+    reader.readData()
+    val sentences = ListBuffer[SpRLSentence]()
+    reader.documents.asScala.foreach(doc => {
+      val sentenceOffsetList = getDocumentSentences(doc, version)
+      sentenceOffsetList.foreach(s => {
+          val ta = TextAnnotationFactory.createTextAnnotation(version, doc.getFilename + s._2, s._1)
+          sentences += new SpRLSentence(s._2, ta.sentences.get(0), getRelations(doc, s._2).asJava)
+      })
+    })
+```
+`SpRLDataReader` is the xml data reader which loads data from the dataset. `getDocumentSentences` splits the sentences and returns a list of tuples. Each tuple contains a sentence and it's character offset in the document. 
+`TextAnnotationFactory.createTextAnnotation` uses the NLP-Pipeline to generate feature rich `TextAnnotation` for each sentence. `getRelations` returns the relations that correspond to the current sentence. 
+
+
+### Defining the `DataModel`
+In order to identify spatial relations, the authors of [2] used a simple method.
+A set of triples used to represent the spatial relations. They generated many candidates and trained a classifier to classify them. So here is the [`DataModel`](SpRLDataModel.scala) :
+```scala
+  // data model
+  val sentences = node[SpRLSentence]
+  val relations = node[SpRelation]
+  val sentencesToRelations = edge(sentences, relations)
+```
+In this model, we have a set of sentences, each sentence connects to many relations through `sentencesToRelations` edges, which some of them can be `GOLD` or positive spatial relation and others that are `CANDIDATE` or non-spatial relations.
```

Comment:
"which some of them"  --> of course "them" refers to "relations" (and not "sentences"), but it is slightly ambiguous. Could you make it explicit?